ocaml/database/redo_log.ml

```diff
@@ -595,6 +600,12 @@ let shutdown log =
     (* ignore any errors *)
   )
 
+let locked f =
+  let finally () = Mutex.unlock shutdown_m in
```

Comment:
I think we have a Mutex.execute somewhere in Threadext

Reply:
```
let shutdown log =
  Xapi_stdext_threads.Threadext.Mutex.execute shutdown_m (fun () ->
      shutdown' log
  )
```

Reply:
It's define in line 18 and ready to be used:

```
let with_lock = Xapi_stdext_threads.Threadext.Mutex.execute
```